### ISSUE TITLE
Special Price not showing on Facebook Catalog

### ISSUE THREAD

**Issue by CPR-head** [facebookincubator/Facebook-For-OpenCart#388]
Hi, the Special Price not showing on Facebook Catalog.

**Comment by shue249** [comment]
@CPR-head We have some FAQs on special price on this link (look at point 2). https://github.com/facebookincubator/Facebook-For-OpenCart/blob/master/FAQ.md#product-configurations-and-settings

We have a few similar cases earlier, eg https://github.com/facebookincubator/Facebook-For-OpenCart/issues/359. You may want to refer to the thread to see if it helps you.

If all else fails, can you share with us a few such products where the special price is not showing and we can help you investigate further. We will need you to provide us the URL of these products on your website.

**Comment by CPR-head** [comment]
ill give you a few examples
https://mcgorisks.ie/index.php?route=product/product&product_id=7493
![screen 1](https://user-images.githubusercontent.com/64903944/81416038-16d23c80-9141-11ea-90e5-359db1b95f61.png)
![screen 3](https://user-images.githubusercontent.com/64903944/81416044-18036980-9141-11ea-8ad5-a89cab05fdaf.png)
![screen 2](https://user-images.githubusercontent.com/64903944/81416046-189c0000-9141-11ea-8be1-99a11e5c7039.png)

**Comment by CPR-head** [comment]
it also doesnt bring the correct brand info over

**Comment by CPR-head** [comment]
this might be the problem the sales date are all like this

Sale Price End Date | 1970-01-30T23:59+00:00
-- | --
Sale Price Start Date | 1970-01-30T00:00+00:00

**Comment by CPR-head** [comment]
![image](https://user-images.githubusercontent.com/64903944/81499012-35127680-92c0-11ea-8564-267b582de86a.png)

**Comment by shue249** [comment]
@CPR-head We checked on the feed file generated and the sales period and sales price is not set in the feed file, hence you are not seeing the discounts. 

1. Want to check if you are using the OpenCart default product management module.
2. What is the time set in your server, is the time up to date?

**Comment by CPR-head** [comment]
yes im using the default product module, im not sure what you mean about the time set is the server

**Comment by shue249** [comment]
@CPR-head 

We are using the time set in the server to determine if we should be using the discount prices. Here is the code pointer - https://github.com/facebookincubator/Facebook-For-OpenCart/blob/master/upload/system/library/facebookproductformatter.php#L272

In layman terms, this code condition checks if the product discount end time is set after the server time. For example, in your product, you set the end date as 2030-12-31. If your server time is set as beyond 2030, we will not be picking up the special price to be used as discount.

In terms of next steps, we can suggest:
1. You can check the server time to see if it is correctly set
2. If you have developer resources, you can consider removing the condition check to see if it will help solve the problem. There are 2 condition checks, make sure to remove the 1st condition for the check on server time while still keeping the 2nd condition check.

**Comment by CPR-head** [comment]
if (!ini_get('date.timezone')) {
	date_default_timezone_set('Europe/Dublin');
thats the part where the time is set, and as far i know that is correct

can you tell me exactly which condition check i should remove?

**Comment by shue249** [comment]
@CPR-head We are referring to line 272 on the file facebookproductformatter.php. Remove the condition "strtotime($sale_end) >= time()"

![image](https://user-images.githubusercontent.com/11957585/81796214-b4fa4400-953f-11ea-8b2e-62541f8af95f.png)

**Comment by CPR-head** [comment]
i removed that line, but its still the same

**Comment by CPR-head** [comment]
this is the error i am getting now
![image](https://user-images.githubusercontent.com/64903944/81833661-e5d67b00-9537-11ea-949a-348e2dcbb0f7.png)

**Comment by CPR-head** [comment]
when i remove that line, my back end of my website went down. i added back in and the site is back up again

**Comment by shue249** [comment]
Sorry for not communicating correctly.

The exact changes is to replace lines 272 and 273 with this line:

if (strtotime($sale_end) >= strtotime($sale_start)) {

An image to better illustrate this:
![image](https://user-images.githubusercontent.com/11957585/81880885-30e8a080-95c1-11ea-9316-12c403bde85a.png)

**Comment by CPR-head** [comment]
its still the same

**Comment by shue249** [comment]
We will be closing out this ticket. Feel free to reach to us if you have further questions.

**Comment by kreatifmerkezi** [comment]
Hello, I have the same problem, I did what you said, my problem continues. Can you help me.

Please
